linalg/solvers.cpp

```diff
@@ -199,7 +199,16 @@ void OperatorJacobiSmoother::Setup(const Vector &diag)
    const double delta = damping;
    auto D = diag.Read();
    auto DI = dinv.Write();
-   MFEM_FORALL(i, height, DI[i] = delta / D[i]; );
+   const bool use_abs_diag_ = use_abs_diag;
+   MFEM_FORALL(i, height,
+   {
+      if (D[i] == 0.0)
+      {
+         MFEM_ABORT_KERNEL("Zero diagonal entry in OperatorJacobiSmoother");
+      }
+      if (!use_abs_diag_) { DI[i] = delta / D[i]; }
+      else                { DI[i] = delta / std::abs(D[i]); }
```

Comment:
Does it make sense to use `fabs` instead of `std::fabs` here?

Reply:
This is `std::abs` not `std::fabs`. In principle, they should be the same, however, I think I've heard arguments that compilers can handle `std::abs(double)` and `std::fabs(double)` differently but I'm not sure if that's really the case. And if there's a difference, I'm not sure which one is better.

If the question is about the `std::` prefix, I've seen issues before with some compilers where `abs()` is resolved as `abs(int)`, so I always prefer to use `std::abs`. In the case of `fabs`, the `std::` prefix is not necessary, as far as I know.

If you have a preference, feel free to change it -- I just don't think we should use `abs()` without `std::`.

Reply:
I was just wondering if the CUDA compiler will handle `std::abs`, since I think `fabs` is the intrinsic in the CUDA math API.

Reply:
I have a slight preference for `fabs`, but I am pretty sure `std::abs` should be OK too.

Reply:
The CUDA tests worked fine on both Lassen and John's machine, so `std::abs` seems fine -- I think the code path `use_abs_diag_ == true` is tested in the newly updated `{p}mesh-optimizer` miniapps.